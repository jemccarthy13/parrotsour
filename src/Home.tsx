import React, { Suspense, lazy } from "react"

import { Route } from "react-router"
import { HashRouter } from "react-router-dom"

import "./css/snackbar.css"
import "./css/styles.css"
import "./css/body.css"
import "./css/fonts.css"

const ParrotSour = lazy(() => import("./pscomponents/parrotsour"))

import ChangeLog from "./changelog/changelog"

/**
 * This is the main entry point into the front-facing application.
 *
 * The application is loaded via 'chunks' (Googe: webpack code-splitting), but
 * once the application is loaded up front, there is no loading time latency.
 *
 * Once procedural is running, the home page allows selection between procedural
 * and intercept.
 *
 * (TODO -- PROC -- figure out how to default intercept? and provide "Home" btn?)
 */
export default class Home extends React.PureComponent {
  getPS = (): JSX.Element => {
    return (
      <ParrotSour
        type="chooser"
        interceptLink="/#/intercept.html"
        proceduralLink="/#/procedural.html"
      />
    )
  }

  getPSP = (): JSX.Element => {
    return <ParrotSour type="procedural" />
  }

  getPSC = (): JSX.Element => {
    return <ParrotSour type="close" />
  }

  getPSI = (): JSX.Element => {
    return <ParrotSour type="intercept" />
  }

  render(): React.ReactElement {
    return (
      <div className="app">
        <div className="body-content" style={{ width: "100%" }}>
          <HashRouter>
            <Suspense fallback={<div>Loading...</div>}>
              <Route exact path="/" component={this.getPSI} />
              <Route path="/changelog.html" component={ChangeLog} />
              <Route path="/changelog" component={ChangeLog} />
              <Route path="/parrotsour.html" render={this.getPS} />
              <Route path="/parrotsour" render={this.getPS} />
              <Route path="/intercept.html" render={this.getPSI} />
              <Route path="/intercept" render={this.getPSI} />
              <Route path="/close.html" render={this.getPSC} />
              <Route path="/close" render={this.getPSC} />
              <Route path="/procedural.html" render={this.getPSP} />
              <Route path="/procedural" render={this.getPSP} />
            </Suspense>
          </HashRouter>
        </div>
      </div>
    )
  }
}
