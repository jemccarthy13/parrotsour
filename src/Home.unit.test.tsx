import React from "react"
import { mount, shallow } from "enzyme"

import Home from "./Home"
import ParrotSour from "./pscomponents/parrotsour"
import { createHashHistory } from "history"
import ParrotSourIntercept from "./pscomponents/intercept/parrotsourintercept"
import ParrotSourProcedural from "./pscomponents/procedural/parrotsourprocedural"
import ParrotSourChooser from "./pscomponents/parrotsourchooser"

describe("Home", () => {
  beforeAll(() => {
    console.warn(
      "These tests do not accurately render home.\r\n" +
        "Waiting for the Suspense to resolve is not currently implemented/supported"
    )
  })

  it("should_render_default", () => {
    console.warn("This test assumes default is intercept -- should it be?")
    const home = shallow(<Home />)
    expect(home.find(ParrotSour)).toBeDefined()
    expect(home.find(ParrotSourIntercept)).toBeDefined()
    expect(home).toMatchSnapshot()
  })

  it("should_render_with_procedural", async () => {
    const history = createHashHistory()
    history.push("/procedural.html")
    const home = mount(<Home />)
    expect(home.find(ParrotSourProcedural)).toBeDefined()
  })

  it("should_render_with_intercept", async () => {
    const history = createHashHistory()
    history.push("/intercept.html")
    const home = mount(<Home />)
    expect(home.find(ParrotSourIntercept)).toBeDefined()
  })

  it("should_render_with_chooser", async () => {
    const history = createHashHistory()
    history.push("/parrotsour.html")
    const home = mount(<Home />)
    expect(home.find(ParrotSourChooser)).toBeDefined()
  })
})
