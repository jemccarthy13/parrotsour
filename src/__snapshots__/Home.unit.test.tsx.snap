// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should_render_default 1`] = `ShallowWrapper {}`;
